### ISSUE TITLE
Set checkpoints of simulations depending on job runtime

### ISSUE THREAD

**Issue by n01r** [ComputationalRadiationPhysics/picongpu#653]
It would probably be nice to have the opportunity to write checkpoints depending on **job runtime** and not only **number of steps** in the simulation. 

Especially after the implementation of ionization the time needed to calculate a step can probably vary quite much.

**Comment by ax3l** [comment]
job runtime is an interesting idea, but implementing a general [signal](https://github.com/ComputationalRadiationPhysics/picongpu/issues/19) and an abstract simulation control interface could be useful, too (e.g., the batch system could be configured to send a signal 15 min before the WT)

**Comment by bussmann** [comment]
VERY NICE idea to allow an abstract interface.

**Comment by bussmann** [comment]
P.S.: This goes into the direction of "communicating" with the sim by having a sim server where such services could hook into.

**Comment by ax3l** [comment]
yes exactly, basically the whole `simulationControl` needs an abstraction so `signals` #19, sockets and python bindings can hook into.
